utils/tx/bank_send.go

```diff
@@ -0,0 +1,169 @@
+package tx
+
+import (
+	"errors"
+	"math/big"
+	"strconv"
+
+	sdkmath "cosmossdk.io/math"
+	tmbytes "github.com/cometbft/cometbft/libs/bytes"
+	tmtypes "github.com/cometbft/cometbft/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/haqq-network/haqq/utils"
+	evmtypes "github.com/haqq-network/haqq/x/evm/types"
+)
+
+func BankSendTxAsEthereumTx(tx sdk.Tx) (*evmtypes.MsgEthereumTx, error) {
+	if tx == nil {
+		return nil, errors.New("tx is nil")
+	}
+
+	if len(tx.GetMsgs()) == 0 {
+		return nil, errors.New("tx has no messages")
+	}
+
+	if len(tx.GetMsgs()) > 1 {
+		return nil, errors.New("tx has more than one message")
+	}
+
+	msg := tx.GetMsgs()[0]
+	sendMsg, ok := msg.(*banktypes.MsgSend)
+	if !ok {
+		return nil, errors.New("tx message is not a bank.MsgSend")
+	}
```

Comment:
this assumes that only one `banktypes.MsgSend` can be present in a transaction - to be sure of this it would probably make sense to add a blocker for such a transaction in the ante handlers. Did you already add that?